cachito/workers/pkg_managers/gomod.py

```diff
@@ -94,11 +94,14 @@ def contains_package(parent_name: str, package_name: str) -> bool:
     :param package_name: name of package to check
     :return: True if package belongs to parent, False otherwise
     """
-    if not package_name.startswith(parent_name):
+    versionless_parent_name = MODULE_VERSION_RE.sub("", parent_name)
+    versionless_package_name = MODULE_VERSION_RE.sub("", package_name)
```

Comment:
Isn't it enough to remove the `/v\d+$` from the parent name? 

```
>>> r = re.compile(r"/v\d+$")
>>> parent = r.sub("", "go.etcd.io/etcd/v3")
>>> "go.etcd.io/etcd/api/v3".startswith(parent)
True
```

Reply:
+1, I think it is. Another question: how are packages from versioned modules reported? 
* go.etcd.io/etcd/some-package
* go.etcd.io/etcd/some-package/v3
* go.etcd.io/etcd/v3/some-package

I guess it's not the third form, otherwise the current approach would already work. But that raises the question - how should we match packages to parent modules if the project uses more than one major version of the same module? E.g. go.etcd.io/etcd/v3, go.etcd.io/etcd/v2 - how do we decide where some-package belongs?

Reply:
> Isn't it enough to remove the /v\d+$ from the parent name?

Updated, thanks

> how are packages from versioned modules reported?

There are only 2 types of internal packages:
- go.etcd.io/etcd/some-package
- go.etcd.io/etcd/some-package/v3

>how should we match packages to parent modules if the project uses more than one major version of the same module? E.g. go.etcd.io/etcd/v3, go.etcd.io/etcd/v2 - how do we decide where some-package belongs?

I see 3 options: 
1. We can ignore versions of a parent module. We need to check only that the internal package is a part of the parent module. If a parent module v1 doesn't have an internal package X, but a parent module v2 does, and a user made a mistake with versions, it will be his fault and problem :D 
2. We can revert [this commit](https://github.com/release-engineering/cachito/commit/6ce92db8d4b93fe70cece554fc5c71e63b9b79b2). And we will need to specify versions as well
3. We can provide some kind of logic for the validation, but I don't know how to do it yet

Reply:
For allowing local replacements, this change should be perfect. I was thinking about the other places where this function is used, such as the match_parent_module() function which is used in [content manifest code](https://github.com/release-engineering/cachito/blob/8091e35befb9026e43069faa7c11c01e20279398/cachito/web/content_manifest.py#L101). If we match on the versionless module name, we could end up matching packages to incorrect modules.

From a quick check, to me it seems that packages and modules are versioned differently. I cloned https://github.com/etcd-io/etcd and did a few basic checks:
```shell
$ go list -find ./...
go.etcd.io/etcd/v3
go.etcd.io/etcd/v3/contrib/lock/client
go.etcd.io/etcd/v3/contrib/lock/storage
go.etcd.io/etcd/v3/contrib/raftexample
go.etcd.io/etcd/v3/tools/benchmark
go.etcd.io/etcd/v3/tools/benchmark/cmd
go.etcd.io/etcd/v3/tools/etcd-dump-db
go.etcd.io/etcd/v3/tools/etcd-dump-logs
go.etcd.io/etcd/v3/tools/etcd-dump-metrics
go.etcd.io/etcd/v3/tools/local-tester/bridge

$ sed --quiet '/replace (/,/)/p' go.mod 
replace (
	go.etcd.io/etcd/api/v3 => ./api
	go.etcd.io/etcd/client/pkg/v3 => ./client/pkg
	go.etcd.io/etcd/client/v2 => ./client/v2
	go.etcd.io/etcd/client/v3 => ./client/v3
	go.etcd.io/etcd/etcdctl/v3 => ./etcdctl
	go.etcd.io/etcd/etcdutl/v3 => ./etcdutl
	go.etcd.io/etcd/pkg/v3 => ./pkg
	go.etcd.io/etcd/raft/v3 => ./raft
	go.etcd.io/etcd/server/v3 => ./server
	go.etcd.io/etcd/tests/v3 => ./tests
)

$ cd api/
$ go list -find ./...
go.etcd.io/etcd/api/v3/authpb
go.etcd.io/etcd/api/v3/etcdserverpb
go.etcd.io/etcd/api/v3/etcdserverpb/gw
go.etcd.io/etcd/api/v3/membershippb
go.etcd.io/etcd/api/v3/mvccpb
go.etcd.io/etcd/api/v3/v3rpc/rpctypes
go.etcd.io/etcd/api/v3/version
go.etcd.io/etcd/api/v3/versionpb
```
So it appears that:
* the main module is `go.etcd.io/etcd/v3`
  * all of its packages are `go.etcd.io/etcd/v3/...`
* there is a bunch of submodules (!= subpackages), such as `go.etcd.io/etcd/api/v3`
  * all the packages of the api submodule are `go.etcd.io/etcd/api/v3/...`

Reply:
TLDR: for matching packages to parent modules, the version should *not* be stripped. Packages still have to start with the entire module name, even if the module name contains a version.

The version should only be stripped when checking local replacements. I would suggest to do it outside the `contains_package()` function

Reply:
I had a bad feeling this change could impact other parts of the code. Thanks for the explanation, @chmeliik++.

Reply:
You are right. I changed it now. Made it 2 line change instead of one to make it easier to follow. :D

Reply:
Let me change the tests